src/test/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImplTest.java

```diff
@@ -385,10 +381,10 @@ protected void additionalSetupForFileProcessing() throws IOException, ParseExcep
         fileNames.add(EMPTY_DATA_FILE);
         fileNames.add(GOOD_DATA_FILE_2);
 
-        BatchInputFileService batchInputFileService = EasyMock.createMock(CuBatchInputFileServiceImpl.class);
-        EasyMock.expect(batchInputFileService.listInputFileNamesWithDoneFile(EasyMock.isA(BatchInputFileType.class))).andReturn(fileNames);
-        EasyMock.expect(batchInputFileService.parse(EasyMock.isA(BatchInputFileType.class), EasyMock.isA(byte[].class))).andReturn(setupAchIncomeFilesGood()).andThrow(new RuntimeException()).andReturn(setupAchIncomeFilesGood());
-        EasyMock.replay(batchInputFileService);
+        BatchInputFileService batchInputFileService = mock(CuBatchInputFileServiceImpl.class);
+        when(batchInputFileService.listInputFileNamesWithDoneFile(isA(BatchInputFileType.class))).thenReturn(fileNames);
+        when(batchInputFileService.parse(isA(BatchInputFileType.class), isA(byte[].class))).thenReturn(setupAchIncomeFilesGood()).thenThrow(new RuntimeException()).thenReturn(setupAchIncomeFilesGood());
```

Comment:
similar to above, we may want to switch this to use Mockito.any here.

Also, how does putting the thenReturn/thenThrow calls work here?

Reply:
first call will return setupAchIncomeFilesGood(), next call will throw exception, subsequent call will return setupAchIncomeFilesGood()

Reply:
Interesting, so that makes a call order dependency to be aware of, but cool technique.  Thank you.